pkg/cloud/network.go

```diff
@@ -58,9 +58,38 @@ func (c *client) ResolveNetwork(csCluster *infrav1.CloudStackCluster) (retErr er
 	return nil
 }
 
+func generateNetworkTagName(csCluster *infrav1.CloudStackCluster) string {
+	return clusterTagNamePrefix + string(csCluster.UID)
+}
+
 func (c *client) GetOrCreateNetwork(csCluster *infrav1.CloudStackCluster) (retErr error) {
-	if retErr = c.ResolveNetwork(csCluster); retErr == nil { // Found network.
+	// Tag the network so we can clean it up later
+	tagNetwork := func(csCluster *infrav1.CloudStackCluster, addCreatedByTag bool) error {
```

Comment:
Not entirely sure why we're defining the func on the fly. Was there some closure goodness here that was needed?

Reply:
Just because it's not needed outside this scope, and it started out as a really simple function.  But now that's is grown, it makes more sense to define it outside the function.  I'll move it.